### PR TITLE
add event when a build references an input imagestreamtag local ref p…

### PR DESCRIPTION
**Pull request by gabemontero**
…olicy and the image registry is not set on imagestream

@openshift/openshift-team-developer-experience FYI / PTAL if you have the cycles 

/assign @bparees 

@bparees - this is the attempt to add some non-intrusive / fairly qualified debug data around builds referencing imagestreams via local policy refs as part of their input before the image registry URL is available so that pullthrough can occur

re: https://bugzilla.redhat.com/show_bug.cgi?id=1721380#c11 and https://bugzilla.redhat.com/show_bug.cgi?id=1721380#c13

### CODE REVIEW

File: pkg/build/controller/build/build_controller.go [hunk 1]
```diff
@@ -959,6 +959,22 @@ func resolveImageStreamTag(ref *corev1.ObjectReference, lister imagev1lister.Ima
...
+		// b) tag is local reference
+		// c) status DockerImageRepository is not set, meaning we got this copy of the stream from the watch
+		//    before the image registry was set up
+		// note, this method is only called for non-output image refs, so we should not get duplicates with
+		// the error/warning currently in place for the output link falling into this category
+		ref2, ok2 := imageutil.SpecHasTag(stream, tag)
+		if ok2 && ref2.ReferencePolicy.Type == imagev1.LocalTagReferencePolicy && len(stream.Status.DockerImageRepository) == 0 {
+			bc.recorder.Eventf(build, corev1.EventTypeNormal,
+				"LocalRefImageStreamTagUnavailable",
+				"The build %s:%s references an image stream tag %s with the local reference policy type, but the image registry location was not marked in the image stream",
```
Comment (bparees) [inline pkg/build/controller/build/build_controller.go]:
```suggestion
				"The build %s/%s references an image stream tag %s with the local reference policy type, but the internal image registry location was not marked in the image stream",
```

**Comment by bparees** [comment]
/approve

one nit and lgtm

**Comment by gabemontero** [comment]
suggested commit accepted and squashed @bparees PTAL

**Pull request merged**